### PR TITLE
fix: provides a fix for handling external HTTP backends

### PR DESCRIPTION
**Pull request by kotharironak**
## Description
Addresses the issue - https://github.com/hypertrace/hypertrace/issues/235
This was happening after a fix of service call related - https://github.com/hypertrace/hypertrace-ingester/pull/159

Reason:
The endpoint enricher enriches all the events with API_ATTRIBUTE_ID associated with the given ApiNode. However, BackendEntityEnricher was using existence check for the same attribute to handle partial trace scenario for HTTP/GRPC backend. However, this condition is not handling the proper scenario.

Solution:
As destination API called by the client service will be associated with a service. In the case of a broken (or partial trace) scenario, we are missing links, so we are missing the entry span of the called service. And that service could be,
- identified by either HTTP BackendResolver or GRPC backend resolver 
- or span's attributes related to peer service

We are using the above information to fix this.

### Testing
locally using docker-compose setup

### Checklist:
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by skjindal93 — COMMENTED** [review]
Can you please add unit tests for the same?

**Comment by kotharironak** [comment]
> Can you please add unit tests for the same?

Done.

**Review by skjindal93 — DISMISSED** [review]

File: hypertrace-trace-enricher/hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/BackendEntityEnricher.java [hunk 1]
```diff
@@ -96,29 +96,23 @@ private boolean isValidBackendEntity(Event backendSpan, BackendInfo candidateInf
...
+    if (checkIfServiceEntityExists(backendSpan.getCustomerId(), fqn, candidateInfo.getEntity())) {
+      return false;
     }
 
     // if there's no child, but, it is a partial trace, then check if the destination API
-    // is internal to the application
-    return !SpanAttributeUtils.containsAttributeKey(
-        backendSpan, EntityConstants.getValue(ApiAttribute.API_ATTRIBUTE_ID));
-    // if it couldn't find a child that's not a service
+    // And, as destination API can be associated either with service identified by backend resolver
```
Comment (tim-mwangi) [inline hypertrace-trace-enricher/hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/BackendEntityEnricher.java:105]: Can this comment be fixed up?

Comment (kotharironak) [reply hypertrace-trace-enricher/hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/BackendEntityEnricher.java]: Done

**Review by tim-mwangi — APPROVED** [review]

**Review by JBAhire — APPROVED** [review]

**Review by skjindal93 — APPROVED** [review]

**Pull request merged**